husky_components/navigation/main.js

```diff
@@ -830,61 +812,69 @@ define(function() {
+            // if a main navigation point should be selected, select its first sub-point
+            if ($selectItem.hasClass('js-navigation-item')) {
+                $selectItem = $(item).closest('li');
             }
 
-            item = this.getItemById(this.sandbox.dom.data($subItem, 'id'));
-
-            // if toggle was clicked, do not set active and selected
-            if (this.sandbox.dom.hasClass($parent, 'navigation-items-toggle') || this.sandbox.dom.hasClass($parent, 'navigation-subitems-toggle')) {
+            // if clicked item is already selected do nothing
+            if ($selectItem.hasClass('is-selected')) {
                 return;
             }
-
-            if (!item.event) {
-                this.sandbox.dom.removeClass(this.sandbox.dom.find('.is-selected', this.$el), 'is-selected');
-                this.sandbox.dom.addClass($subItem, 'is-selected');
-
-                this.sandbox.dom.removeClass(this.sandbox.dom.find('.is-active', this.$el), 'is-active');
-                this.sandbox.dom.addClass($items, 'is-active');
+            // if a toggelable item was clicked nothing should be selected, it just toggles
+            if ($(item).parent().hasClass('navigation-items-toggle') || $(item).parent().hasClass('navigation-subitems-toggle')) {
+                return;
             }
 
-            var extendedItem = item;
+            itemData = this.getItemById(this.sandbox.dom.data($selectItem, 'id'));
+            extendedItem = this.sandbox.util.extend(true, {}, itemData, options);
 
-            if (!!options) {
-                extendedItem = this.sandbox.util.extend(true, {}, extendedItem, options);
+            // If it is a navigation-point with a custom configured event (like actions), the event gets emitted and nothing else.
+            if (!!itemData.event && emit !== false) {
+                this.emitItemSelectEvent(extendedItem);
+                return;
             }
 
             if (emit !== false) {
-                if (item.event) {
-                    this.sandbox.emit('husky.navigation.item.event.' + item.event, item.eventArgs);
-                } else {
-                    this.sandbox.emit(EVENT_ITEM_SELECT, extendedItem);
-                }
+                this.emitItemSelectEvent(extendedItem);
             }
 
-            if (this.hasDataNavigation()) {
-                this.removeDataNavigation();
-            }
+            this.markItemSelected($selectItem);
+            this.refreshDataNavigation(extendedItem);
+        },
+
+        /**
+         * Marks a given navigation item as selected. The main navigation-point corresponding
+         * to the given item gets marked as active
+         * @param $item The item to select
+         */
+        markItemSelected: function($item) {
+            this.unmarkSelectedItem();
+            this.sandbox.dom.addClass($item, 'is-selected');
+            this.sandbox.dom.addClass($item.closest('.js-navigation-items'), 'is-active');
```

Comment:
You can also use jQuery directly here, as you've already changed in other parts of this PR.